File: UI/src/app/shared/models/identity-card.ts
import { employee } from "./employee";

export interface WorkPermitDetail {
  id: string;
  permitType: string;
  permitStartDate: string; // Change to appropriate date type if needed
  permitExpiryDate: string; // Change to appropriate date type if needed
  document: string;
}

export interface IdentityCard {
  id: string;
  passport: string;
  visaStartDate: string; // Change to appropriate date type if needed
  visaExpiryDate: string; // Change to appropriate date type if needed
  visa: string;
  blueCardStartDate: string; // Change to appropriate date type if needed
  blueCardExpiryDate: string; // Change to appropriate date type if needed
  blueCard: string;
  employeeId: string; // Assuming string type for foreign key, change to appropriate type if needed

  action: number;
  workPermitDetails?: WorkPermitDetail[];
}

// "id": "7bcb7af4-871a-4b7b-1ac8-08dc789aef7f",
// "passport": "string",
// "visaStartDate": "1998-01-01",
// "visaExpiryDate": "1998-01-01",
// "visa": "string",
// "blueCardStartDate": "1998-01-01",
// "blueCardExpiryDate": "1998-01-01",
// "blueCard": "string",
// "employeeId": "bc0425a8-34cd-46ec-9cb8-2eda5192a747",

File: API/beONHR.Entities/IdentityCard.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace beONHR.Entities
{
    public class IdentityCard
    {
        [Key]
        public Guid Id { get; set; }
        public string Passport { get; set; }
        public DateOnly VisaStartdate { get; set; }
        public DateOnly VisaExpirytdate { get; set; }
        public string Visa { get; set; }
        public DateOnly BlueCardStartdate { get; set; }
        public DateOnly BlueCardExpirytdate { get; set; }
        public string BlueCard { get; set; }

        public Guid EmployeeId { get; set; } // foreign key
        public bool IsDeleted { get; set; } = false;
        [ForeignKey("EmployeeId")]
        public virtual Employee? Employee { get; set; }
        public virtual ICollection<WorkPermitDetail> WorkPermitDetails { get; set; }
    }
}

namespace beONHR.Entities
{
    public class WorkPermitDetail
    {
        [Key]
        public Guid Id { get; set; }
        public string PermitType { get; set; }
        public DateOnly PermitStartdate { get; set; }
        public DateOnly PermitExpirytdate { get; set; }
        public Guid IdentityId { get; set; }//foriegnkey
        public string Document { get; set; }

        [ForeignKey("IdentityId")]
        public virtual IdentityCard? identity { get; set; }

    }
}


